tests/Integration/ScriptsTest.php

```diff
@@ -334,131 +272,134 @@ public function test_user_logged_in(): void {
+		// First user is not logged-in to the second blog, so track_authenticated_users value
+		// is irrelevant. Confirm that tracker script is registered and enqueued.
+		$this->assert_script_statuses(
+			'wp-parsely-tracker',
+			array( 'enqueued', 'registered' )
 		);
 	}
 
 	/**
-	 * Test the wp_parsely_enqueue_js_tracker filter
-	 * When it returns false, the tracking script should not be enqueued.
+	 * Test that the tracker script is correctly output in HTML markup
+	 * when the wp_parsely_enable_cfasync_attribute filter is used.
 	 *
 	 * @covers \Parsely\Scripts::enqueue_js_tracker
+	 * @uses \Parsely\Parsely::get_asset_cache_buster
 	 * @uses \Parsely\Parsely::api_key_is_missing
 	 * @uses \Parsely\Parsely::api_key_is_set
 	 * @uses \Parsely\Parsely::get_options
 	 * @uses \Parsely\Parsely::post_has_trackable_status
 	 * @uses \Parsely\Parsely::update_metadata_endpoint
+	 * @uses \Parsely\Scripts::register_scripts
+	 * @uses \Parsely\Scripts::script_loader_tag
+	 * @group scripts
+	 * @group scripts-output
 	 */
-	public function test_enqueue_js_tracker_filter(): void {
-		add_filter( 'wp_parsely_enqueue_js_tracker', '__return_false' );
+	public function test_tracker_markup_has_attribute_when_cfasync_filter_is_used(): void {
+		add_filter( 'wp_parsely_enable_cfasync_attribute', '__return_true' );
 
 		ob_start();
-		$post_array = $this->create_test_post_array();
-		$post       = $this->factory->post->create( $post_array );
-		$this->go_to( '/?p=' . $post );
+		$this->go_to_new_post();
+		self::$scripts->register_scripts();
 		self::$scripts->enqueue_js_tracker();
-		$intermediate_output = ob_get_contents();
-
-		self::assertSame(
-			'',
-			$intermediate_output,
-			'Failed to confirm scripts were not printed by enqueue_js_tracker()'
-		);
 
 		wp_print_scripts();
-
 		$output = ob_get_clean();
+
 		self::assertSame(
-			'',
+			// phpcs:ignore WordPress.WP.EnqueuedResources.NonEnqueuedScript
+			"<script data-cfasync=\"false\" type='text/javascript' data-parsely-site=\"blog.parsely.com\" src='https://cdn.parsely.com/keys/blog.parsely.com/p.js?ver=" . Parsely::VERSION . "' id=\"parsely-cfg\"></script>\n",
 			$output,
-			'Failed to confirm filter prevented enqueued scripts'
+			'Tracker script tag was not printed correctly'
 		);
 	}
+
+	/**
+	 * Assert multiple enqueueing statuses for a script.
+	 *
+	 * @param string $handle       Script handle to test.
+	 * @param array  $assert_true  Optional. Statuses that should assert to true. Accepts 'enqueued', 'registered', 'queue', 'to_do', and 'done'. Default is an empty array.
+	 * @param array  $assert_false Optional. Statuses that should assert to false. Accepts 'enqueued', 'registered', 'queue', 'to_do', and 'done'. Default is an empty array.
+	 *
+	 * @throws RiskyTestError If no assertions ($assert_true, $assert_false) get passed to the function.
+	 */
+	public function assert_script_statuses( string $handle, array $assert_true = array(), array $assert_false = array() ): void {
```

Comment:
I think we can change the type declarations from `array` to the wider `iterable` (PHP 7.1). We're only looping through the collections; we're not using access the values through array syntax. This potentially gives other tests the ability to use anything that implements Traversable

```suggestion
	public function assert_script_statuses( string $handle, iterable $assert_true = array(), iterable $assert_false = array() ): void {
```

If it still works, then the docs need updating as well.

I'm rusty on whether `count()` will work on a Traversable / Iterator though, or if there would be another way to detect that the object was empty of values - if that makes it messy, ignore this suggestion and stick with `array`.

Reply:
I checked on this and would prefer to stick with `array`. From my research, `Traversable` doesn't automatically imply `Countable` (only some `Traversable` objects implement `Countable`). So adding `iterable` would imply a broader scope of possibilities and introduce more complexity. Using `iterator_to_array()` within the function probably defeats the purpose, and empty() has its own issues and would probably need array conversion as well. So I wouldn't go forward with this unless we were to find a really good reason for it.